mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/DelegatingMailboxMessage.java

```diff
@@ -112,6 +113,10 @@ public MessageId getMessageId() {
         return message.getMessageId();
     }
 
+    public ThreadId getThreadId() {
+        return (ThreadId) message.getMessageId();
```

Comment:
I would not be surprised that this breaks if you try to really access it... Did you see the implementations of MessageId for memory and Cassandra?